src/browser-driver-manager.js

```diff
@@ -1,259 +1,124 @@
 
-    child.stdout.on('data', chunk => {
-      lines.push(chunk);
-
-      if (!returnValue) {
-        process.stdout.write(chunk);
-      }
-    });
-
-    // for some reason the install script will output curl download
-    // percent through stderr
-    child.stderr.on('data', chunk => {
-      // mimic download percent bar by manually clearing lines
-      // @see https://stackoverflow.com/questions/17309749/node-js-console-log-is-it-possible-to-update-a-line-rather-than-create-a-new-l
-      if (chunk.indexOf('\r') !== -1) {
-        process.stdout.clearLine();
-      }
-      if (lastChunk) {
-        process.stdout.cursorTo(lastChunk.length - 1);
-      }
-      process.stdout.write(chunk);
+  if (!args[0]) {
+    throw new Error(
+      'Please specify browser and version in browser@version format'
+    );
+  }
 
-      lastChunk = chunk;
+  // When parsing the values the version value could be set
+  // as a version number (e.g. 116.0.5845.96) or a channel (e.g. beta, dev, canary)
+  const [browser, version = 'latest'] = args[0].split('@');
 
-      if (chunk.toString().indexOf('browser-driver-manager error') !== -1) {
-        process.exitCode = 1;
-        reject();
-      }
-    });
-    child.on('close', () => {
-      if (returnValue) {
-        // output all logs except the last one
-        for (let i = 0; i < lines.length - 1; i++) {
-          process.stdout.write(lines[i]);
-        }
-        resolve(lines[lines.length - 1].toString());
-      } else {
-        resolve();
-      }
-    });
-  });
-}
+  // Create a cache directory if it does not exist on the user's home directory
+  // This will be where environment variables will be stored for the tests
+  // since it is a consistent location across different platforms
+  console.log(`BDM_CACHE_DIR: ${BDM_CACHE_DIR}`);
+  if (!fs.existsSync(BDM_CACHE_DIR)) {
+    await fsPromises.mkdir(BDM_CACHE_DIR, { recursive: true });
+  }
 
-/**
- * Install ChromeDriver from npm
- * @param {String} version - Version to install
- * @param {Boolean} verbose - If verbose logging is enabled
- */
-function installChromeDriver(version, verbose) {
-  return new Promise(resolve => {
-    const child = spawn('npm', [
-      'install',
-      '--no-save',
-      `chromedriver@${version}`
-    ]);
-    if (verbose) {
-      child.stdout.on('data', chunk => {
-        process.stdout.write(chunk);
-      });
-    }
-    child.stderr.on('data', chunk => {
-      process.stderr.write(chunk);
+  if (browser.includes('chrome')) {
+    const platform = detectBrowserPlatform();
+    // This will sync the browser and chromedriver versions
+    const chromeBuildId = await resolveBuildId(
+      Browser.CHROME,
+      platform,
+      version
+    );
+
+    const installChrome = await installBrowser(
+      BDM_CACHE_DIR,
+      Browser.CHROME,
+      chromeBuildId
+    );
+
+    const installChromedriver = await installBrowser(
+      BDM_CACHE_DIR,
+      Browser.CHROMEDRIVER,
+      chromeBuildId
+    );
```

Comment:
Why are you running the exact same code as the `installBrowser` function does? Couldn't you just call `installBrowser` passing it the version directly?

```suggestion
    const installChrome = await installBrowser(
      BDM_CACHE_DIR,
      Browser.CHROME,
      version
    );

    const installChromedriver = await installBrowser(
      BDM_CACHE_DIR,
      Browser.CHROMEDRIVER,
      version
    );
```